lib/open_api_spex/cast/array.ex

```diff
@@ -16,7 +17,43 @@ defmodule OpenApiSpex.Cast.Array do
 
   ## Private functions
 
-  defp cast_items(%{value: items} = ctx) do
+  defp cast_array(%{value: value, schema: %{minItems: minimum}} = ctx)
+       when is_list(value) and is_integer(minimum) do
```

Comment:
`value` is guaranteed to be a list, because of `cast/1`, above. You can remove the guard to remove the extra noise.